.github/workflows/lint.yml

```diff
@@ -1,4 +1,7 @@
+# The goal of this workflow is mostly to run pre-commit on every pull requests.
+
 name: lint
+
```

Comment:
I haven't touched much of this workflow, will try and answer questions but may leave gaps based on context.

Reply:
@brendongo might have context.